lib/ClangImporter/ImportType.cpp

```diff
@@ -1287,10 +1287,19 @@ static ImportedType adjustTypeForConcreteImport(
   // we would prefer to instead use the default Swift convention.
   if (hint == ImportHint::Block) {
     if (canBridgeTypes(importKind)) {
+      // Determine the representation we need. For Objective-C generic
+      // arguments, we cannot bridge them to a block, so force a block
+      // representation even if our imported type thus far is a Swift
+      // function representation.
+      auto requiredFunctionTypeRepr = FunctionTypeRepresentation::Swift;
+      if (importKind == ImportTypeKind::ObjCGenericArgument) {
+        requiredFunctionTypeRepr = FunctionTypeRepresentation::Block;
+      }
+
       auto fTy = importedType->castTo<FunctionType>();
       FunctionType::ExtInfo einfo = fTy->getExtInfo();
-      if (einfo.getRepresentation() != FunctionTypeRepresentation::Swift) {
-        einfo = einfo.withRepresentation(FunctionTypeRepresentation::Swift);
+      if (einfo.getRepresentation() != requiredFunctionTypeRepr) {
+        einfo = einfo.withRepresentation(requiredFunctionTypeRepr);
```

Comment:
Since `@convention(block)` is the default, it seems simple to just change the condition here: `canBridgeTypes(importKind) && importKind != ImportTypeKind::ObjCGenericArgument`.

Reply:
This is the rathole I went down before, because the way in which we pass down the desired "bridgeability" of the imported type doesn't account for blocks, which are bridgeable in some positions but not others. The problem with simply trying to exclude `ObjCGenericArgument` here is that the `mappedType` we get here has already been given the Swift function type representation (because we're importing a `typedef` with "full bridgeability"), and we no longer have the block type around.

Reply:
Ah, typedefs. In that case, can you update the comment above?

Reply:
Yeah.